src/System.IO.FileSystem/src/System/IO/FindEnumerableFactory.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO
+{
+    internal static class FindEnumerableFactory
+    {
+        internal static void NormalizeInputs(ref string directory, ref string expression)
+        {
+            if (PathHelpers.IsPathRooted(expression))
+                throw new ArgumentException(SR.Arg_Path2IsRooted, nameof(expression));
+
+            // We always allowed breaking the passed in directory and filter to be separated
+            // any way the user wanted. Looking for "C:\foo\*.cs" could be passed as "C:\" and
+            // "foo\*.cs" or "C:\foo" and "*.cs", for example. As such we need to combine and
+            // split the inputs if the expression contains a directory separator.
+            //
+            // We also allowed for expression to be "foo\" which would translate to "foo\*".
+
+            ReadOnlySpan<char> directoryName = PathHelpers.GetDirectoryNameNoChecks(expression.AsReadOnlySpan());
+
+            if (directoryName.Length != 0)
+            {
+                // Need to fix up the input paths
+                directory = PathHelpers.CombineNoChecks(directory, directoryName);
+                expression = expression.Substring(directoryName.Length + 1);
```

Comment:
Can we use spans rather than strings and slice instead of substring?

Reply:
We could, but this allocation should be pretty rare.